HW_1/task_3.py

```diff
@@ -0,0 +1,26 @@
+# Напишите код, который запрашивает число и сообщает является ли оно простым или составным.
+# Используйте правило для проверки: “Число является простым, если делится нацело только на единицу и на себя”.
+# Сделайте ограничение на ввод отрицательных чисел и чисел больше 100 тысяч.
+from math import sqrt
+
+MIN_NUM = 1
+MAX_NUM = 100000
+START = 2
+result = ""
+
+while True:
+    num = int(input('Введите число от 1 до 100 000: '))
+    if MIN_NUM <= num <= MAX_NUM:
+        for i in range(START, int(num ** 0.5) + 1):
+            if num % i == 0:
+                result = f'Число {num} является составным'
+                break
+            else:
+                result = f'Число {num} является простым'
```

Comment:
В эту строку код может зайти несколько раз, следовательно, несколько раз произойдет изменение адреса в памяти для переменной `result`,  а это лишние ресурсы процессора и памяти. Почему бы не использовать связку `break` и `else` от цикла?
```suggestion
        else:
            result = f'Число {num} является простым'
```
Понятны ли вам исправления?